docs/pages/api-docs/data-grid.md

```diff
@@ -66,6 +66,37 @@ import { DataGrid } from '@material-ui/data-grid';
 | <span class="prop-name">sortModel</span> | <span class="prop-type">SortModel</span> |   | Set the sort model of the grid. |
 
 The `ref` is forwarded to the root element.
+## Slots
+
+Api of the `components` props of type `GridSlotsComponent`
+
+| Name | Type | Default | Description |
+|:-----|:-----|:--------|:------------|
+| <span class="prop-name">ColumnMenu</span> | <span class="prop-type">React.ElementType&lt;ColumnMenuProps></span> | <span class="prop-type">GridColumnMenu</span> | Column menu component rendered by clicking on the 3 dots "kebab" icon in column headers.|
+| <span class="prop-name">ColumnsPanel</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">ColumnsPanel</span> | Columns panel component rendered when clicking the columns button.|
+| <span class="prop-name">ErrorOverlay</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps & ErrorOverlayProps></span> | <span class="prop-type">ErrorOverlay</span> | Error overlay component rendered above the grid when an error is caught.|
+| <span class="prop-name">FilterPanel</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">FilterPanel</span> | Filter panel component rendered when clicking the filter button.|
+| <span class="prop-name">Footer</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">GridFooter</span> | Footer component rendered at the bottom of the grid viewport.|
+| <span class="prop-name">Header</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">GridHeader</span> | Header component rendered above the grid column header bar.|
+| <span class="prop-name">LoadingOverlay</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">LoadingOverlay</span> | Loading overlay component rendered when the grid is in a loading state.|
+| <span class="prop-name">NoRowsOverlay</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">NoRowsOverlay</span> | No rows overlay component rendered when the grid has no rows.|
+| <span class="prop-name">Pagination</span> | <span class="prop-type">React.ElementType&lt;BaseComponentProps></span> | <span class="prop-type">Pagination</span> | Pagination component rendered in the grid footer by default.|
+| <span class="prop-name">Panel</span> | <span class="prop-type">React.ElementType<BaseComponentProps & PanelProps></span> | <span class="prop-type">Panel</span> | Panel component wrapping the filters and columns panels. |
+
+### Icons Slots
+
+| Name | Type | Default | Description |
+|:-----|:-----|:--------|:------------|
```

Comment:
I don't think that it makes sense to differentiate icons as such, cc @mbrookes 

```suggestion
```

Reply:
What's the alternative? (I'm assuming we do want to make these customizable?)

Reply:
One alternative is to treat icons as any other components.

Reply:
Having the tables separated does have the benefit of not cluttering one with the other, but no strong preference.

Reply:
My main question is: Why discriminate icons over other types of components?

Reply:
Nothing, it's just easier to mentally process two short lists than one long one, and in this case icons seem like a natural grouping. But merged is fine if you prefer.

Reply:
Ok, no strong preferences. I guess we can still automate the split by checking that the slot name ends with `Icon`.

Reply:
I think where it might break down is if there are only one or two icons, or it's a short list to start with. Then what do you do? Keep a single list (inconsistent), or always have two (may not be beneficial)?